components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.hpp

```diff
@@ -565,6 +565,44 @@ static void rrtmgp_sw(
   int ngpt = k_dist.get_ngpt();
   int ngas = gas_concs.get_num_gases();
 
+  // Allocate temporaries from pool
+  const int size1 = nday;
+  const int size2 = nday*nlay; // 4
+  const int size3 = nday*(nlay+1); // 5
+  const int size4 = ncol*nlay;
+  const int size5 = nbnd*nday; //2
+  const int size6 = nday*ngpt;
+  const int size7 = nday*(nlay+1)*nbnd; // 3
+  const int size8 = ncol*nlay*(k_dist.get_ngas()+1);
+
+  RealT* data = pool_t::template alloc_raw<RealT>(size1 + size2*4 + size3*5 + size4 + size5*2 + size6 + size7*3 + size8), *dcurr = data;
```

Comment:
From what I can tell, rrtmgp_sw is called by rrtmgp_main, so we'll still be allocating the pool at runtime. Is that correct? Can we maybe allocate the pool once at init, pass it around at runtime, and deallocate it inside finalize?